src/checks/hyper-training-31.js

```diff
@@ -0,0 +1,14 @@
+module.exports = {
+  description: 'A maxed IV (31) cannot be hypertrained.',
+  field: 'iv',
+  test (pkmn) {
+    return !(
+      pkmn.ivHp === 31 && pkmn.hyperTrainedHp ||
```

Comment:
Rework the logic on this one as well, please. It's also really confusing to do.

Fitler on gen 7, and make it similar to how I did the other one.